corehq/apps/reports/standard/monitoring.py

```diff
@@ -1820,8 +1820,11 @@ def user_rows(self):
             self.domain, self.request.GET.getlist(EMWF.slug), self.request.couch_user
         )
         chunk_size = 50000
-        user_iterator = user_es_query.size(chunk_size).scroll()
-        for user_chunk in chunked(user_iterator, chunk_size):
+        first_id_in_chunk = 0
+        user_chunk_total = user_es_query.fields(util.SimplifiedUserInfo.ES_FIELDS).size(chunk_size).run().total
+        while first_id_in_chunk < user_chunk_total:
+            user_chunk = user_es_query.fields(util.SimplifiedUserInfo.ES_FIELDS).start(first_id_in_chunk).size(
+                chunk_size).run().hits
```

Comment:
The problem with this is that if it takes a long time to process, users could be added or removed, which would throw off these indices, possibly causing this to skip users or include duplicates.  Scroll queries avoid this problem.  The same issue occurred for exports: https://github.com/dimagi/commcare-hq/pull/20282

You could try using `iter_es_docs_from_query`, which uses a scroll query to get a list of ids up front, streams that list of IDs to disk (in case it's too big to hold in memory), and then pulls the documents in chunks as needed.  Individual documents may change throughout the duration of the query, but the set of documents returned should be fixed.

Reply:
Implemented here: https://github.com/dimagi/commcare-hq/pull/23782/commits/a900e5f50299dd463958631ef0d14dfaf61772da